client/components/woopay/save-user/checkout-page-save-user.js

```diff
@@ -193,7 +200,11 @@ const CheckoutPageSaveUser = ( { isBlocksCheckout } ) => {
 			return;
 		}
 
-		if ( isSaveDetailsChecked && ! isPhoneValid ) {
+		if (
+			isSaveDetailsChecked &&
```

Comment:
It seems like `isSaveDetailsChecked` is truthy even when the Save My Info component is not rendered. Making it return `false` when the component is not visible is likely another solution for the issue.

Reply:
@malithsen do you think it makes sense to change [these lines](https://github.com/Automattic/woocommerce-payments/blob/4b687737240f7fd19e5cb5ef2addf0d6473e835e/client/components/woopay/save-user/checkout-page-save-user.js#L37-L39) to also take into consideration if `isWCPayChosen` ?

Reply:
I tested this solution but at that point `isWCPayChosen` is set to false which prevents the checkbox from being pre checked for non $0 carts.